journal/src/main/scala/com/evolutiongaming/kafka/journal/ConsumerPool.scala

```diff
@@ -0,0 +1,80 @@
+package com.evolutiongaming.kafka.journal
+
+import cats.effect.Resource
+import cats.effect.kernel.{Async, Ref}
+import cats.effect.std.Queue
+import cats.effect.syntax.resource._
+import cats.syntax.all._
+import com.evolutiongaming.catshelper.{Log, LogOf}
+import com.evolutiongaming.kafka.journal.Journals.Consumer
+
+/**
+ * A fixed size pool.
+ *
+ * Semantically blocks on `acquire` if there are no consumers available.
+ *
+ * Make sure to release all consumers before shutting the pool down.
+ */
+private[journal] trait ConsumerPool[F[_]] {
+
+  def acquire: F[Consumer[F]]
```

Comment:
why do you need such api? can't we just keep using `Resource[Consumer]` ?

Reply:
Yeah, good point, I should probably use Resource instead